pynest/examples/astrocytes/astrocyte_brunel_fixed_indegree.py

```diff
@@ -0,0 +1,374 @@
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Random balanced network with astrocytes
+---------------------------------------
+
+This script simulates a random balanced network with excitatory and inhibitory
+neurons and astrocytes. The astrocytes are modeled with ``astrocyte_lr_1994``,
+implemented according to [1]_, [2]_, and [3]_. The neurons are modeled with
+``aeif_cond_alpha_astro``, an adaptive exponential integrate-and-fire neuron
+supporting neuron-astrocyte interactions.
+
+The simulation results show how astrocytes affect neuronal excitability. The
+figures displayed at the end of the simulation show the astrocytic dynamics,
+the slow inward current induced by the astrocytes in the postsynaptic neurons,
+and a raster plot of neuronal firings, respectively.
+
+In this version of the model, primary connections between populations are
+created with the fixed-indegree rule.
+
+References
+~~~~~~~~~~
+
+.. [1] Li, Y. X., & Rinzel, J. (1994). Equations for InsP3 receptor-mediated
+       [Ca2+]i oscillations derived from a detailed kinetic model: a
+       Hodgkin-Huxley like formalism. Journal of theoretical Biology, 166(4),
+       461-473. DOI: https://doi.org/10.1006/jtbi.1994.1041
+
+.. [2] De Young, G. W., & Keizer, J. (1992). A single-pool inositol
+       1,4,5-trisphosphate-receptor-based model for agonist-stimulated
+       oscillations in Ca2+ concentration. Proceedings of the National Academy
+       of Sciences, 89(20), 9895-9899. DOI:
+       https://doi.org/10.1073/pnas.89.20.9895
+
+.. [3] Nadkarni, S., & Jung, P. (2003). Spontaneous oscillations of dressed
+       neurons: a new mechanism for epilepsy?. Physical review letters, 91(26),
+       268101. DOI: https://doi.org/10.1103/PhysRevLett.91.268101
+
+See Also
+~~~~~~~~
+
+:doc:`astrocyte_small_network`
+
+"""
+
+###############################################################################
+# Import all necessary modules for simulation and plotting.
+
+import random
+
+import matplotlib.pyplot as plt
+import nest
+import numpy as np
+
+###############################################################################
+# Set simulation parameters.
+
+sim_params = {
+    "dt": 0.1,  # simulation resolution in ms
+    "sim_time": 1000.0,  # simulation time in ms
+    "N_rec_spk": 100,  # number of neurons to record from with spike recorder
+    "N_rec_mm": 50,  # number of nodes (neurons, astrocytes) to record from with multimeter
+    "n_threads": 4,  # number of threads for NEST
```

Comment:
I changed it here to n_vp as well. Users should think mainly in terms of virtual processes, since NEST guarantees identical results for simulations run with a fixed number of VPs, independent of how they are split into MPI processes and threads. That split is rather a technicality.

Reply:
Okay, thanks for the insight